### PR TITLE
chore: seed less random tags

### PR DESCRIPTION
**Pull request by Sboonny**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

clear task in #1339

I am planning to use event tags for chapter tags, but chapter should have three tags or maybe different tags? idk after the mvp

### CODE REVIEW

**Review by ojeytonwilliams — COMMENTED** [review]
This looks good, but is it necessary to reduce the number of tags?

**Comment by Sboonny** [comment]
at first I though I could swap `create` with `createMany` and add multiple tag.

```js
          createMany: {
            tag: {
              connectOrCreate: {
                create: { name: tagNames[randomTag] },
                where: { name: tagNames[randomTag] },
              },
            },
            tag: {
              connectOrCreate: {
                create: { name: tagNames[randomTag] },
                where: { name: tagNames[randomTag] },
              },
            },
          },
```

but that didn't work, so I pushed to create style and planning to change it later. 

If you have suggestion of what I can do, I am open for it

**Comment by ojeytonwilliams** [comment]
The issue is that `createMany` doesn't [accept nested inputs ](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany), so you can't use `connectOrCreate` inside of it.

You could create all the tags and then connect them to each event, but it's easier to use `connectOrCreate` inside the `prisma.events.update` call.

Something like this, if you always want 3 tags.

```ts
    const tags = randomItems(tagNames, 3, true);
    const connectOrCreateTags = tags.map((tag) => ({
      tag: {
        connectOrCreate: {
          where: { name: tag },
          create: { name: tag },
        },
      },
    }));

    await prisma.events.update({
      where: { id: event.id },
      data: {
        tags: {
          create: connectOrCreateTags,
        },
      },
    });
```

File: server/prisma/generator/factories/events.factory.ts [hunk 1]
```diff
@@ -107,19 +107,23 @@ const createEvents = async (
       'Ruby',
       'Rust',
     ];
-    const randomTag = Math.floor(Math.random() * tagNames.length);
+    const randomNum = Math.round((Math.random() * tagNames.length) / 5);
+
+    const tags = randomItems(tagNames, randomNum, true);
```
Comment (Sboonny) [inline server/prisma/generator/factories/events.factory.ts:112]:
```suggestion
    const tagsAmount = Math.round((Math.random() * tagNames.length) / 5);

    const tags = randomItems(tagNames, tagsAmount, true);
```

File: server/prisma/generator/factories/events.factory.ts [hunk 2]
```diff
@@ -84,19 +84,46 @@ const createEvents = async (
...
+
+    const tags = randomItems(tagNames, tagsAmount, true);
+    const connectOrCreateTags = tags.map((tag) => ({
+      tag: {
+        connectOrCreate: {
+          where: { name: tag },
+          create: { name: tag },
+        },
+      },
+    }));
```
Comment (ojeytonwilliams) [inline server/prisma/generator/factories/events.factory.ts:120]:
```suggestion
    const tagNames = randomItems(tagNames, tagsAmount, true);
    const connectOrCreateTags = tagNames.map((name) => ({
      tag: {
        connectOrCreate: {
          where: { name },
          create: { name },
        },
      },
    }));
```
I know I suggested the first way of naming it, but this seems nicer!

File: server/prisma/generator/factories/events.factory.ts [hunk 3]
```diff
@@ -84,19 +84,46 @@ const createEvents = async (
...
+      'Svelete',
+      'SveleteKit',
+      'Vite',
+      'Prisma',
+      'Ruby',
+      'Rust',
+    ];
+    const tagsAmount = Math.round((Math.random() * tagNames.length) / 5);
+
+    const selectedTags = randomItems(tagNames, tagsAmount, true);
```
Comment (gikf) [inline server/prisma/generator/factories/events.factory.ts:112]:
```suggestion
    const tagsCount = Math.round((Math.random() * tagNames.length) / 5);

    const selectedTags = randomItems(tagNames, tagsCount, true);
```

Nitpicking

**Pull request merged**